### ISSUE TITLE
Support 409 for PATCH requests.

### ISSUE THREAD

**Issue by elenacanovi** [clojure-liberator/liberator#312]
As per the RFC specification (https://tools.ietf.org/html/rfc5789#section-2.2), `PATCH` requests should be able to return 409 in the cases of conflicting state or concurrent modification (my use case btw). However, if I understand correctly the decision graph, this possibility is currently not supported by Liberator if `method-patch?` is `true`.

As a possible solution, I tried to add a new decision called `patch-to-existing?` to replace `method-patch?` in the `false` branch of `method-delete?` and moved `method-patch?` to the `false` branch of `conflict?`. If  `method-patch?` is true then the path leads to `patch!`, otherwise it leads to `method-post?` from where the old behaviour is recovered.

I committed this variant to my fork here: https://github.com/elenacanovi/liberator/commit/87dc3e5e4255ccd8d644657fa2b1a35c0df671a3.

**Comment by ordnungswidrig** [comment]
Thanks, that looks like a missing feature. For make this go into liberator's code base we need tests and docs, too. Please tell me if you want to have a take on it.

**Comment by elenacanovi** [comment]
Yes, I would be glad to. Is it ok if I prepare a pull request with also test and docs?

**Comment by ordnungswidrig** [comment]
@elenacanovi Sure! I would love that.

**Comment by casseds2** [comment]
Hi guys, was there ever any follow up on this issue?

**Comment by elenacanovi** [comment]
Not from my side, sorry. I'm currently not working on that.

**Comment by ordnungswidrig** [comment]
@casseds2 I'll try to find the time to merge it. As writren above this requires updates to the documentation and tests which takes some sime.

**Comment by casseds2** [comment]
Thank you very much Philipp! You're help is greatly appreciated :ok_hand: :slightly_smiling_face:
